websocket.go

```diff
@@ -278,60 +278,101 @@ outerReadLoop:
 		var tun *StreamingTunnel
 		path := strings.TrimPrefix(r.URL.Path, "/"+t.serviceName+"/")
 		log.Lvlf2("ws request from %s: %s/%s", r.RemoteAddr, t.serviceName, path)
-		reply, tun, err = s.ProcessClientRequest(r, path, buf)
-		if err == nil {
-			if tun == nil {
-				tx += len(reply)
-				if err = ws.SetWriteDeadline(time.Now().Add(5 * time.Minute)); err != nil {
-					log.Error(err)
-					break
+
+		isStreaming := false
+		bidirectionalStreamer, ok := s.(BidirectionalStreamer)
+		if ok {
+			isStreaming, err = bidirectionalStreamer.IsStreaming(path)
+			if err != nil {
+				log.Errorf("failed to check if it is a streaming "+
+					"request %s/%s: %+v", t.serviceName, path, err)
+				continue
```

Comment:
Each of these continue/breaks could use a comment explaining the reasoning for why it has to be that. I suspect that this part of the code is pretty lightly (or not at all?) covered by unit tests. What does code coverage say?

This is an important place to be careful, because getting the outer loop into an endless loop would be very bad here: hung grouting in a spin loop, possibly. (Been there, done that, got the coredump.)

Reply:
Agreed. It follows the same error handling as what has already been implemented: if err is not nil then the loop ends. 
(Coverage: ~85%)